src/KubeOps.Abstractions/Reconciliation/ReconciliationResult{TEntity}.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+
+using k8s;
+using k8s.Models;
+
+namespace KubeOps.Abstractions.Reconciliation;
+
+/// <summary>
+/// Represents the result of an operation performed on an entity
+/// within the context of Kubernetes controllers or finalizers.
+/// </summary>
+/// <typeparam name="TEntity">
+/// The type of the Kubernetes entity associated with this result.
+/// Must implement <see cref="IKubernetesObject{TMetadata}"/> where TMetadata is <see cref="V1ObjectMeta"/>.
+/// </typeparam>
+public sealed record ReconciliationResult<TEntity>
+    where TEntity : IKubernetesObject<V1ObjectMeta>
+{
+    private ReconciliationResult(TEntity entity, bool isSuccess, string? errorMessage, Exception? error, TimeSpan? requeueAfter)
+    {
+        Entity = entity;
+        IsSuccess = isSuccess;
+        ErrorMessage = errorMessage;
+        Error = error;
+        RequeueAfter = requeueAfter;
+    }
+
+    /// <summary>
+    /// Represents the Kubernetes entity associated with the result of an operation or reconciliation process.
+    /// This property contains the entity object of type <typeparamref name="TEntity"/> that was processed, modified, or finalized
+    /// during an operation. It provides access to the updated state or metadata of the entity after the operation.
+    /// Typically used for handling further processing, queuing, or logging of the affected entity.
+    /// </summary>
+    public TEntity Entity { get; }
+
+    /// <summary>
+    /// Indicates whether the operation has completed successfully.
+    /// Returns true when the operation was successful, and no errors occurred.
+    /// This property is used to determine the state of the operation
+    /// and is often checked to decide whether further processing or error handling is necessary.
+    /// When this property is true, <see cref="ErrorMessage"/> will be null, as there were no errors.
+    /// </summary>
+    [MemberNotNullWhen(false, nameof(ErrorMessage))]
+    public bool IsSuccess { get; }
+
+    /// <summary>
+    /// Indicates whether the operation has failed.
+    /// Returns true when the operation was unsuccessful, as represented by <see cref="IsSuccess"/> being false.
+    /// Typically used to distinguish failure states and to conditionally handle error scenarios such as logging or retries.
+    /// When this property is true, <see cref="ErrorMessage"/> will be non-null, providing additional details about the error.
+    /// </summary>
+    [MemberNotNullWhen(true, nameof(ErrorMessage))]
+    public bool IsFailure => !IsSuccess;
```

Comment:
Since this is a "copy paste" of `IsSuccess` above, but just the negation: is it necessary? People can just check "if not Success". And we would have a property less on the class, to keep it simple. But open to suggestions.